napari/layers/base/base.py

```diff
@@ -256,6 +257,7 @@ def __init__(
         self.events = EmitterGroup(
             source=self,
             auto_connect=True,
+            event_added=self._connect_event,
```

Comment:
Instead of adding a new keyword argument to `EmitterGroup` to connect each `EventEmitter` in the `EmitterGroup` to `Layer.clean_cache`, could we connect the group itself?

```python
self.events.connect(self.clean_cache)
```

and then modify `clean_cache` to skip any events of type `'status'`?

E.g. something like: https://github.com/napari/napari/blob/8c6630131f4ea60084410e998bcf919bcdfc2a69/napari/_vispy/layers/points.py#L155

I think this also prevents deprecated events from being emitted too.

Alternatively, could we be more selective and only connect to events that we think will change the extent (e.g. data, transforms)? Or is that too risky?

Reply:
> Alternatively, could we be more selective and only connect to events that we think will change the extent (e.g. data, transforms)? Or is that too risky?

The only problem with a limited list is that someone could add something in the future which will change the behavior of the extent and refresh of the cache will not happen. So maybe better will be to introduce a list of events that should not execute clean. Then all new events will execute clean and need to be manually excluded. 

Also, switch to `psygnal` may need to force us to use another approach.